helpdesk_mgmt_sla_rule/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class HelpDeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    sla_id = fields.Many2one("helpdesk.sla", string="SLA", compute="_compute_sla")
+
+    def check_sla_rule_m2o(self, rule, check, field):
+        if not rule[field] or not check:
+            return True
+        elif self[field].id == rule[field].id:
+            return True
+        else:
+            return False
```

Comment:
```suggestion
        return (not rule[field] or not check) or (self[field].id == rule[field].id)
```
this should work to replace your tests